### ISSUE TITLE
Regression rendering HasMany fields

### ISSUE THREAD

**Issue by kberzinch** [laravel/nova-issues#1388]
- Laravel Version: v5.6.39
- Nova Version: v1.3.1
- PHP Version: 7.2.15
- Operating System and Version: Ubuntu 18
- Browser type and version: Chrome 72

### Description:
Something changed between v1.3.0 (works) and v1.3.1 (doesn't work) that broke one of the HasMany fields in my application. (v1.3.2 also does not work.) This one field uses a different foreign key instead of the model's ID. Possibly related to #109?

Field in Nova:
```php
HasMany::make('Attendance')->canSee(function ($request) {
    if ($request->resourceId == $request->user()->id) {
        return $request->user()->can('read-attendance-own');
    } else {
        return $request->user()->can('read-attendance');
    }
}),
```

Relationship on model:
```php
public function attendance()
{
    return $this->hasMany(\App\Attendance::class, 'gtid', 'gtid');
}
```

Error from AJAX call during page load:
```
BadMethodCallException thrown with message "Method Illuminate\Database\Query\Builder::getForeignKey does not exist."
```

<details>
  <summary>Stacktrace:</summary>
<pre>
#89 BadMethodCallException in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php:2816
#88 Illuminate\Database\Query\Builder:__call in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php:1286
#87 Illuminate\Database\Eloquent\Builder:__call in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Relations/Relation.php:369
#86 Illuminate\Database\Eloquent\Relations\Relation:__call in /var/www/apiary/staging/vendor/laravel/nova/src/Fields/ResolvesReverseRelation.php:66
#85 Laravel\Nova\Fields\BelongsTo:Laravel\Nova\Fields\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Support/Arr.php:175
#84 call_user_func in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Support/Arr.php:175
#83 Illuminate\Support\Arr:first in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Support/Collection.php:657
#82 Illuminate\Support\Collection:first in /var/www/apiary/staging/vendor/laravel/nova/src/Fields/ResolvesReverseRelation.php:67
#81 Laravel\Nova\Fields\BelongsTo:getReverseRelation in /var/www/apiary/staging/vendor/laravel/nova/src/Fields/ResolvesReverseRelation.php:30
#80 Laravel\Nova\Fields\BelongsTo:isReverseRelation in /var/www/apiary/staging/vendor/laravel/nova/src/Fields/BelongsTo.php:137
#79 Laravel\Nova\Fields\BelongsTo:isNotRedundant in /var/www/apiary/staging/vendor/laravel/nova/src/Fields/BelongsTo.php:122
#78 Laravel\Nova\Fields\BelongsTo:authorize in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Support/HigherOrderCollectionProxy.php:60
#77 Illuminate\Support\HigherOrderCollectionProxy:Illuminate\Support\{closure} in [internal]:0
#76 array_filter in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Support/Arr.php:606
#75 Illuminate\Support\Arr:where in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Support/Collection.php:471
#74 Illuminate\Support\Collection:filter in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Support/HigherOrderCollectionProxy.php:61
#73 Illuminate\Support\HigherOrderCollectionProxy:__call in /var/www/apiary/staging/vendor/laravel/nova/src/ResolvesFields.php:168
#72 Laravel\Nova\Resource:resolveFields in /var/www/apiary/staging/vendor/laravel/nova/src/ResolvesFields.php:31
#71 Laravel\Nova\Resource:indexFields in /var/www/apiary/staging/vendor/laravel/nova/src/Resource.php:285
#70 Laravel\Nova\Resource:serializeForIndex in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Support/HigherOrderCollectionProxy.php:60
#69 Illuminate\Support\HigherOrderCollectionProxy:Illuminate\Support\{closure} in [internal]:0
#68 array_map in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Support/Collection.php:932
#67 Illuminate\Support\Collection:map in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Support/HigherOrderCollectionProxy.php:61
#66 Illuminate\Support\HigherOrderCollectionProxy:__call in /var/www/apiary/staging/vendor/laravel/nova/src/Http/Controllers/ResourceIndexController.php:24
#65 Laravel\Nova\Http\Controllers\ResourceIndexController:handle in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Controller.php:54
#64 call_user_func_array in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Controller.php:54
#63 Illuminate\Routing\Controller:callAction in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php:45
#62 Illuminate\Routing\ControllerDispatcher:dispatch in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Route.php:212
#61 Illuminate\Routing\Route:runController in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Route.php:169
#60 Illuminate\Routing\Route:run in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Router.php:665
#59 Illuminate\Routing\Router:Illuminate\Routing\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:30
#58 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /var/www/apiary/staging/vendor/laravel/nova/src/Http/Middleware/Authorize.php:18
#57 Laravel\Nova\Http\Middleware\Authorize:handle in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:151
#56 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#55 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /var/www/apiary/staging/vendor/laravel/nova/src/Http/Middleware/BootTools.php:20
#54 Laravel\Nova\Http\Middleware\BootTools:handle in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:151
#53 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#52 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /var/www/apiary/staging/vendor/laravel/nova/src/Http/Middleware/DispatchServingNovaEvent.php:20
#51 Laravel\Nova\Http\Middleware\DispatchServingNovaEvent:handle in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:151
#50 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#49 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php:41
#48 Illuminate\Routing\Middleware\SubstituteBindings:handle in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:151
#47 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#46 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php:68
#45 Illuminate\Foundation\Http\Middleware\VerifyCsrfToken:handle in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:151
#44 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#43 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php:49
#42 Illuminate\View\Middleware\ShareErrorsFromSession:handle in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:151
#41 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#40 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php:63
#39 Illuminate\Session\Middleware\StartSession:handle in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:151
#38 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#37 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php:37
#36 Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse:handle in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:151
#35 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#34 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php:66
#33 Illuminate\Cookie\Middleware\EncryptCookies:handle in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:151
#32 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#31 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /var/www/apiary/staging/app/Http/Middleware/CASAuthenticate.php:63
#30 App\Http\Middleware\CASAuthenticate:handle in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:151
#29 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#28 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:104
#27 Illuminate\Pipeline\Pipeline:then in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Router.php:667
#26 Illuminate\Routing\Router:runRouteWithinStack in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Router.php:642
#25 Illuminate\Routing\Router:runRoute in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Router.php:608
#24 Illuminate\Routing\Router:dispatchToRoute in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Router.php:597
#23 Illuminate\Routing\Router:dispatch in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:176
#22 Illuminate\Foundation\Http\Kernel:Illuminate\Foundation\Http\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:30
#21 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /var/www/apiary/staging/vendor/laravel/nova/src/Http/Middleware/ServeNova.php:26
#20 Laravel\Nova\Http\Middleware\ServeNova:handle in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:151
#19 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#18 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /var/www/apiary/staging/vendor/fideloper/proxy/src/TrustProxies.php:57
#17 Fideloper\Proxy\TrustProxies:handle in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:151
#16 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#15 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php:31
#14 Illuminate\Foundation\Http\Middleware\TransformsRequest:handle in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:151
#13 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#12 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php:31
#11 Illuminate\Foundation\Http\Middleware\TransformsRequest:handle in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:151
#10 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#9 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php:27
#8 Illuminate\Foundation\Http\Middleware\ValidatePostSize:handle in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:151
#7 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#6 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/CheckForMaintenanceMode.php:62
#5 Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode:handle in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:151
#4 Illuminate\Pipeline\Pipeline:Illuminate\Pipeline\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php:53
#3 Illuminate\Routing\Pipeline:Illuminate\Routing\{closure} in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php:104
#2 Illuminate\Pipeline\Pipeline:then in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:151
#1 Illuminate\Foundation\Http\Kernel:sendRequestThroughRouter in /var/www/apiary/staging/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php:116
#0 Illuminate\Foundation\Http\Kernel:handle in /var/www/apiary/staging/public/index.php:53
</pre>
</details>

### Steps To Reproduce:
I can try to make a tiny app to reproduce this in case the above isn't sufficient, just let me know.

**Comment by dkulyk** [comment]
This has been fixed in v1.3.2

**Comment by kberzinch** [comment]
I am still seeing this behavior in v1.3.2.

**Comment by dkulyk** [comment]
May you make an app to reproduce this issue?

**Comment by kberzinch** [comment]
Sure, I have a couple things to push out for the next few days but I'll try to have something by the end of the week.

**Comment by dkulyk** [comment]
what are relations on Attendance resource?

**Comment by kberzinch** [comment]
`Attendance` snippets:

```php
public function attendee()
{
    return $this->belongsTo(\App\User::class, 'gtid', 'gtid');
}

public function attendable()
{
    return $this->morphTo();
}

public function recorded()
{
    return $this->hasOne('\App\User', 'id', 'recorded_by');
}
```

`Attendable`s have
```php
public function attendance()
{
    return $this->morphMany(\App\Attendance::class, 'attendable');
}
```

**Comment by dkulyk** [comment]
there are models. What are fields on Attendance Nova resource?

**Comment by kberzinch** [comment]
Ah, sorry.

```php
BelongsTo::make('User', 'attendee'),

MorphTo::make('Attended', 'attendable')
    ->types([
        Event::class,
        Team::class,
    ]),

(new BelongsTo('Recorded By', 'recorded', 'App\Nova\User'))
    ->help('The user that recorded the swipe'),
```

**Comment by kberzinch** [comment]
Ok, after a lot of staring at `ResolvesReverseRelation::getReverseRelation` and my code, I figured out that the exception was caused by that function attempting to get the foreign key of a `hasOne` relationship. I can patch this in the application by replacing that relationship with a `belongsTo`.

I did try this on a blank Laravel 5.8 + Nova 2.0.0 and wasn't able to get it to break in the same way.
